searx/plugins/tor_check.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""
+A plugin to check if the ip address of the request is a TOR exit node
+if the user searches for "tor" "tor check" or "check tor"
+It fetches the tor exit node list from https://check.torproject.org/exit-addresses
+and parses all the IPs into a list, then checks if the user's ip address
+is in it.
+
+Enable in 'settings.yml' (uncomment "Tor check plugin"
+in the 'enabled_plugins' section)
+"""
+
+import re
+from urllib.request import urlopen
+from urllib.error import HTTPError
+
+name = "Tor check plugin"
+description = """
+This plugin checks if the address of the request is a TOR exit node,
+and informs the user if it is, like check.torproject.org but from searxng.
+"""
+default_on = False
+
+# Regex for exit node addresses in the list.
+reg = re.compile(r"(?<=ExitAddress )\S+")
+
+
+def post_search(request, search):
+
+    if search.search_query.query.lower() in ("tor", "tor check", "check tor"):
```

Comment:
Idea : use the same trigger than the `self_info` plugin.

So the query `ip` (nothing else) triggers `self_info` and this plugin. 
The users can pick the one they want (or both but it is not useful).

Alternative: include this code into `self_info`, but I guess some people don't want to fetch content from `check.torproject.org` when they want their IP address.

---

https://github.com/searxng/searxng/blob/44f2eb50a5566e8e14a1e970f1ed775bf025c1ea/searx/plugins/self_info.py#L37

Reply:
I think it would help with the convenience to keep it like that and on by default, its more intuitive to type "tor" to check if you're using tor and "ip" to just check your IP than having  them both in the same place.

Reply:
I reduced it to ``query_keywords = ['tor-check']``